### PR TITLE
[#17909] fix: unhandled error when app launched in offline mode

### LINKED ISSUE #17909: "java.lang.NullPointerException" error is shown if  app while is relaunched in the flight mode
### Steps:
1. Close the app
2. Turn on the flight mode
3. Open the app again

### Actual result:
Error on the Android "java.lang.NullPointerException" error is shown for Android

image

Error on the IOS "No protocols method IMap. - dissoc defined"

https://github.com/status-im/status-mobile/assets/52490791/759d5944-55f4-455a-8064-0da40190f5e3

### Logs:
[logs (3).zip](https://github.com/status-im/status-mobile/files/13363509/logs.3.zip)

### ENV:
- Nightly: 15 Nov 2023

### PR DESCRIPTION
**Pull request by mohsen-ghafouri**
fixes #17909

### Summary
- Read certificate PEM in offline mode throws null pointer error in android
- timesource had issue when device is offline in status-go, already fixed here [status-go/pull/4304](https://github.com/status-im/status-go/pull/4304). i just pointed the app to latest changes

### Steps to test

1. Close the app
2. Turn on the flight mode
3. Open the app again

### Result

iOS

https://github.com/status-im/status-mobile/assets/71308738/63a5d99f-4306-4d6a-99f6-4594670d397d

status: ready

### CODE REVIEW

**Comment by siddarthkay** [comment]
Hey @mohsen-ghafouri : Thanks for your contribution here, I too recently faced issue getting the android build to work when there was no internet connection.
I'll try out your branch locally and report back.

**Review by siddarthkay — APPROVED** [review]
tested locally, works offline! good job @mohsen-ghafouri

**Review by vkjr — APPROVED** [review]

**Comment by mohsen-ghafouri** [comment]
status-go tag is updated in develop and this PR doesn't require separate branch in go. now it's just mobile changes

**Comment by pavloburykh** [comment]
@mohsen-ghafouri thank you for the PR! I am not reproducing the issue anymore. I also have not reproduced the issue in latest develop thought, as you have mentioned the correspondent go fix is already there. 

@mohsen-ghafouri just want to clarify: do we still need this PR to be merged in mobile? Asking because it seems that bug is not reproducible in develop even without this commit. But I might be missing something.

**Comment by mohsen-ghafouri** [comment]
@pavloburykh In this PR, I've added additional checks to prevent crashes on both Android and iOS platforms. These extra safety measures are beneficial and shouldn't pose any issues.

**Comment by pavloburykh** [comment]
> @pavloburykh In this PR, I've added additional checks to prevent crashes on both Android and iOS platforms. These extra safety measures are beneficial and shouldn't pose any issues.

Great! Thank you for that @mohsen-ghafouri! I have triggered re-run of failed e2e. Let's wait for the results and after that we will be ready to merge. I will ping you here.

**Comment by pavloburykh** [comment]
@mohsen-ghafouri PR is ready for merge. Thank you!

**Comment by mohsen-ghafouri** [comment]
Thank you @pavloburykh

**Pull request merged**
